iothub/device/src/net451/Common/LegacyClientWebSocketTransport.cs

```diff
@@ -245,50 +220,40 @@ protected override bool CloseInternal()
             try
             {
                 if (Logging.IsEnabled)
-                {
                     Logging.Enter(this, $"{nameof(LegacyClientWebSocketTransport)}.{nameof(CloseInternal)}");
-                }
 
-                var webSocketState = this.webSocket.State;
-                if (webSocketState != IotHubClientWebSocket.WebSocketState.Closed && webSocketState != IotHubClientWebSocket.WebSocketState.Aborted)
+                IotHubClientWebSocket.WebSocketState webSocketState = _webSocket.State;
+                if (webSocketState != IotHubClientWebSocket.WebSocketState.Closed
+                    && webSocketState != IotHubClientWebSocket.WebSocketState.Aborted)
                 {
-                    this.CloseInternalAsync();
+                    _ = CloseInternalAsync();
                 }
                 return true;
             }
             finally
             {
                 if (Logging.IsEnabled)
-                {
                     Logging.Exit(this, $"{nameof(LegacyClientWebSocketTransport)}.{nameof(CloseInternal)}");
-                }
-            }            
+            }
         }
 
-        async Task CloseInternalAsync()
+        private async Task CloseInternalAsync()
         {
             try
             {
                 if (Logging.IsEnabled)
-                {
                     Logging.Enter(this, $"{nameof(LegacyClientWebSocketTransport)}.{nameof(CloseInternalAsync)}");
-                }
 
-                await this.webSocket.CloseAsync().ConfigureAwait(false);
+                await _webSocket.CloseAsync().ConfigureAwait(false);
             }
-            catch (Exception e)
+            catch (Exception e) when (!Fx.IsFatal(e))
```

Comment:
Before, we were rethrowing when it's a fatal exception. After, we don't attempt to rethrow if it's a fatal exception. Is this expected?

Reply:
Now we only catch when it isn't fatal. Meaning, if it is fatal, the exception is not caught.

Before, we'd rethrow if it was fatal.

So unless I've made a grievous error, it should be behaving the same.